src/driver/amdxdna/ve2_hwctx.c

```diff
@@ -613,3 +613,209 @@ void ve2_hwctx_fini(struct amdxdna_ctx *hwctx)
 	ve2_free_hsa_queue(xdna, &hwctx->priv->hwctx_hsa_queue);
 	kfree(hwctx->priv);
 }
+
+static int ve2_hwctx_config_dtrace_buf(struct amdxdna_ctx *hwctx, u64 base_paddr,
+				       struct uc_info_entry uc_info[], bool attach)
+{
+	struct device *aie_dev = hwctx->priv ? hwctx->priv->aie_dev : NULL;
```

Comment:
doesn't make sense to me. Why hwctx->priv is NULL? This is something must be initialized in ctx_init() call back. If you do have reason for this line, that must be a bug needs to be fix somewhere else.